sdk/python/kfp/containers/entrypoint.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Union
+
+from absl import logging
+import fire
+from google.protobuf import json_format
+import os
+
+from kfp.containers import _gcs_helper
+from kfp.containers import entrypoint_utils
+from kfp.dsl import artifact
+
+_FN_SOURCE = 'ml/main.py'
+_FN_NAME_ARG = 'function_name'
+
+_PARAM_METADATA_SUFFIX = '_input_param_metadata_file'
+_ARTIFACT_METADATA_SUFFIX = '_input_artifact_metadata_file'
+_FIELD_NAME_SUFFIX = '_input_field_name'
+_ARGO_PARAM_SUFFIX = '_input_argo_param'
+_INPUT_PATH_SUFFIX = '_input_path'
+_OUTPUT_NAME_SUFFIX = '_input_output_name'
+
+_OUTPUT_PARAM_PATH_SUFFIX = '_parameter_output_path'
+_OUTPUT_ARTIFACT_PATH_SUFFIX = '_artifact_output_path'
+
+_METADATA_FILE_ARG = 'executor_metadata_json_file'
+
+
+class InputParam(object):
+  """POD that holds an input parameter."""
+
+  def __init__(self,
+      value: Optional[str] = None,
+      metadata_file: Optional[str] = None,
+      field_name: Optional[str] = None):
+    """Instantiates an InputParam object.
+
+    Args:
+      value: The actual value of the parameter.
+      metadata_file: The location of the metadata JSON file output by the
+        producer step.
+      field_name: The output name of the producer.
+
+    Raises:
+      ValueError: when neither of the following is true:
+        1) value is provided, and metadata_file and field_name are not; or
+        2) both metadata_file and field_name are provided, and value is not.
+    """
+    if not (value is not None and not (metadata_file or field_name) or (
+        metadata_file and field_name and value is None)):
+      raise ValueError('Either value or both metadata_file and field_name '
+                       'needs to be provided. Get value={value}, field_name='
+                       '{field_name}, metadata_file={metadata_file}'.format(
+          value=value,
+          field_name=field_name,
+          metadata_file=metadata_file
+      ))
+    if value:
+      self._value = value
+    else:
+      # Parse the value by inspecting the producer's metadata JSON file.
+      self._value = entrypoint_utils.get_parameter_from_output(
+          metadata_file, field_name)
+
+    self._metadata_file = metadata_file
+    self._field_name = field_name
+
+  # Following properties are read-only
+  @property
+  def value(self) -> Union[float, str, int]:
```

Comment:
Just noticed this return type is union of float, str, int. In that case, should the class constructor also take `value` with the union type?

Reply:
Agree. Changed the type hint